packages/template-retail-react-app/app/contexts/index.js

```diff
@@ -46,6 +46,44 @@ CategoriesProvider.propTypes = {
     categories: PropTypes.object
 }
 
+export const LocaleContext = React.createContext()
+export const LocaleProvider = ({locale: initialLocale = {}, children}) => {
+    const [locale, setLocale] = useState(initialLocale)
+
+    return <LocaleContext.Provider value={{locale, setLocale}}>{children}</LocaleContext.Provider>
+}
+
+LocaleProvider.propTypes = {
+    children: PropTypes.node.isRequired,
+    locale: PropTypes.object
+}
+
+export const SiteContext = React.createContext()
+export const SiteProvider = ({site: initialSite = {}, children}) => {
+    const [site, setSite] = useState(initialSite)
+
+    return <SiteContext.Provider value={{site, setSite}}>{children}</SiteContext.Provider>
+}
+
+SiteProvider.propTypes = {
+    children: PropTypes.node.isRequired,
+    site: PropTypes.object
+}
+
+export const AppConfigContext = React.createContext()
+export const AppConfigProvider = ({urlTemplateLiteral, children}) => {
+    return (
+        <AppConfigContext.Provider value={{urlTemplateLiteral}}>
```

Comment:
Other providers are named after the values they contain. So how about we name `AppConfigProvider` to something else more specific?

Reply:
Also, this would cause a lot of confusion with the existing `AppConfig` component.

Reply:
Given these three providers are used for the same purpose, would it make sense to group them into a single provider?